crates/topos-tce-storage/src/tests/mod.rs

```diff
@@ -401,3 +404,45 @@ async fn get_pending_certificates(storage: RocksDBStorage) {
     );
     assert_eq!(expected_pending_certificates, pending_certificates);
 }
+
+#[rstest]
+#[tokio::test]
+async fn fetch_source_subnet_certificates_in_order(storage: RocksDBStorage) {
+    let mut certificates = create_certificate_chain(SOURCE_SUBNET_ID_1, &[TARGET_SUBNET_ID_1], 10);
+    certificates.swap(1, 2);
+
+    for certificate in certificates {
+        _ = storage.persist(&certificate, None).await;
+    }
+    let (builder, client, _) = Connection::build(Box::pin(async move { Ok(storage) }));
+
+    tokio::spawn(builder.into_future());
+
+    let res = client
+        .fetch_certificates(crate::FetchCertificatesFilter::Source {
+            source_stream_position: crate::CertificateSourceStreamPosition {
+                source_subnet_id: SOURCE_SUBNET_ID_1,
+                position: Position::ZERO,
+            },
+            limit: 100,
+        })
+        .await
+        .unwrap();
+
+    let mut accumulator = 0;
```

Comment:
What is the purpose of this variable? Was the idea to assert on it being `10` at the end of the loop?